tests/frontend/specs/timeslider_follow.js

```diff
@@ -1,55 +1,69 @@
-describe("timeslider", function(){
+describe("timeslider follow", function(){
   //create a new pad before each test run
   beforeEach(function(cb){
     helper.newPad(cb);
-    this.timeout(6000);
   });
 
-  it("follow content as it's added to timeslider", function(done) { // passes
-    var inner$ = helper.padInner$;
-    var chrome$ = helper.padChrome$;
-
-    // make some changes to produce 100 revisions
-    var timePerRev = 900
-      , revs = 10;
-    this.timeout(revs*timePerRev+10000);
-    for(var i=0; i < revs; i++) {
-      setTimeout(function() {
-        // enter 'a' in the first text element
-        inner$("div").last().sendkeys('a\n');
-        inner$("div").last().sendkeys('{enter}');
-        inner$("div").last().sendkeys('{enter}');
-        inner$("div").last().sendkeys('{enter}');
-        inner$("div").last().sendkeys('{enter}');
-      }, timePerRev*i);
+  it("content as it's added to timeslider", async function() {
+    // send 3 revisions
+    let revs = 6;
+    let message = 'a\n\n\n\n\n\n\n\n\n\n';
+    let newLines = message.split('\n').length
+    for (let i=0;i<revs;i++){
+      await helper.edit(message, newLines*i + 1);
     }
 
-    setTimeout(function() {
-      // go to timeslider
-      $('#iframe-container iframe').attr('src', $('#iframe-container iframe').attr('src')+'/timeslider#0');
+    await helper.gotoTimeslider(0);
+    await helper.waitForPromise(function(){return helper.contentWindow().location.hash === '#0'})
 
-      setTimeout(function() {
-        var timeslider$ = $('#iframe-container iframe')[0].contentWindow.$;
-        var $sliderBar = timeslider$('#ui-slider-bar');
+    let originalTop = helper.contentWindow().$('#innerdocbody').offset();
 
-        var latestContents = timeslider$('#innerdocbody').text();
+    // set to follow contents as it arrives
+    helper.contentWindow().$('#options-followContents').prop("checked", true);
+    helper.contentWindow().$('#playpause_button_icon').click();
 
-        // set to follow contents as it arrives
-        timeslider$('#options-followContents').prop("checked", true);
+    let newTop;
+    return helper.waitForPromise(function(){
+      newTop = helper.contentWindow().$('#innerdocbody').offset();
+      return newTop.top < originalTop.top;
```

Comment:
Won't this only wait for a single detected advancement?

Also, if playback is considerably faster than the poll period then this won't detect the intermediate advancements.